libafl_bolts/src/shmem.rs

```diff
@@ -1328,6 +1328,179 @@ pub mod unix_shmem {
             }
         }
     }
+
+    /// Module containing `memfd` shared memory support, usable on Linux and Android.
+    #[cfg(all(unix, feature = "std", not(target_vendor = "apple")))]
+    pub mod memfd {
+        use alloc::string::ToString;
+        use core::{
+            ops::{Deref, DerefMut},
+            ptr, slice,
+        };
+        use std::ffi::CString;
+
+        use libc::{
+            c_void, close, fstat, ftruncate, mmap, munmap, syscall, SYS_memfd_create, MAP_SHARED,
+            PROT_READ, PROT_WRITE,
+        };
+
+        use crate::{
+            shmem::{ShMem, ShMemId, ShMemProvider},
+            Error,
+        };
+
+        /// An memfd based impl for linux/android
+        #[cfg(unix)]
+        #[derive(Clone, Debug)]
+        pub struct MemfdShMem {
+            id: ShMemId,
+            map: *mut u8,
+            map_size: usize,
+        }
+
+        impl MemfdShMem {
+            /// Create a new shared memory mapping, using shmget/shmat
+            pub fn new(map_size: usize) -> Result<Self, Error> {
+                unsafe {
+                    let c_str = CString::new("libAFL").unwrap();
+                    let fd = syscall(SYS_memfd_create, c_str.as_ptr(), 0) as i32;
```

Comment:
Why use a raw syscall here instead of the for example the function from `nix`? 
https://docs.rs/nix/latest/nix/sys/memfd/fn.memfd_create.html